### DISCUSSION TITLE
Error: Connection was closed before we received a valid response from endpoint URL:

### DISCUSSION THREAD

**Discussion by asantoniw** [aws/aws-cli#7370] [Q&A] [answered]
I am getting this error message when trying to use cli to upload files to a s3 bucket. 

`aws s3 cp <file>  s3://lamesatestuno/QBBackups/ `

This cp works fine for smaller files like 19 MB. When I try with larger files I will get the error.

I used to have a a backup working fine with "aws s3 sync ..."for file of 300 MB it did fail with larger files like 3 GB. But now it gives this error even for a 30 MB.

This line with the error appears in line 759 of the Debug

`botocore.exceptions.ConnectionClosedError: Connection was closed before we received a valid response from endpoint URL: "https://lamesatestuno.s3.us-east-1.amazonaws.com/QBBackups/Inmobiliario_.rar?uploadId=hyD2Zo4spyhRJYPViZA6OghkLHe8KNq5wVBuNzLxRhhcHpNSmk_60KSjGATRoUxIQ_wJCLBThuFn7kFUReui4g--&partNumber=2".`

I am attaching the log  created with the --debug option
[Log of error to cp a 30 MB file to s3 bucket.txt](https://github.com/aws/aws-cli/files/9873759/Log.of.error.to.cp.a.30.MB.file.to.s3.bucket.txt)

What would be the problem?

**Comment by RyanFitzSimmonsAK** [comment]
Hi @asantoniw, thanks for reaching out.  I was not able to reproduce this error.  Typically, this is a connectivity issue; could you try making a similar upload from a different device on the same network?  Another thing you could try is increasing the `--cli-connect-timeout`, as noted in this [related issue](https://github.com/aws/aws-cli/issues/3842).  Please let me know if that works, or if you have any other questions.

**Reply by asantoniw** [reply]
@RyanFitzSimmonsAK thanks for the suggestions.
I have tried with the cli-connect-timeout as suggested, set to 6000, and I still have the same error message. 
I will test a similar upload on the same network from another device next week.

**Comment by asantoniw** [comment]
Following @RyanFitzSimmonsAK  advice, installed the CLI on a Mac and tested it from another network worked well.
Now, from the network that presents the "Connection was closed..." I am getting the same response, the error occurs after some 10 MB like originally.
What can be the problem?

**Reply by RyanFitzSimmonsAK** [reply]
Hi @asantoniw.  It could be a problem with your network configuration; are you able to reproduce the issue consistently at an exact file size?  Are you able to consistently upload files smaller than that size?  Also, could you run a speed test on your network and report the results? Finally, can you confirm that us-east-1 is the region you should be using?

**Comment by asantoniw** [comment, marked as answer]
Hi @RyanFitzSimmonsAK we solved the problem with an upgrade to our wan network. The ISP was upgraded to a higher speed (100/50 mbps). 
In this configuration, with no changes to any other parameter and without the cli-connect-timeout flag, I have been able to upload several files of up to 3.5 GB with no problem using 
`aws s3 sync .......`
Thanks for the help.
